sdk/identity/identity/test/httpRequests.ts

```diff
@@ -1,24 +1,37 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
-import http from "http";
-import https from "https";
-import type { AccessToken, GetTokenOptions, TokenCredential } from "../src";
+import { MockInstance, vi } from "vitest";
+
+vi.mock("https", async (importOriginal) => {
```

Comment:
Here's where I try to mock the https and http modules, of course that does not take.

I thought it was a hoisting issue so I moved it higher up but that makes no difference and it sounds like `vi.mock` calls are hoisted already...